### PR TITLE
Add system-account user to nats-server

### PR DESCRIPTION
**Pull request by raypinto**
**Description**

Changes proposed in this pull request:

- This PR is based off of a previously implemented, but rolled backed PR: https://github.com/kyma-project/kyma/pull/15274
- The changes from the previous PR include:
  - Adding the config to the nats-config-reloader
  - Using the right "$SYS" account name as described in the issue: https://github.com/nats-io/k8s/issues/617

**Related issue(s)**

### CODE REVIEW

File: components/eventing-controller/controllers/backend/reconciler.go [hunk 1]
```diff
@@ -458,6 +482,18 @@ func (r *Reconciler) DeletePublisherProxySecret(ctx context.Context) error {
 		Namespace: deployment.PublisherNamespace,
 		Name:      deployment.PublisherName,
 	}
+	return r.deleteSecret(ctx, secretNamespacedName)
+}
+
+func (r *Reconciler) DeleteNATSSecret(ctx context.Context) error {
```
Comment (k15r) [inline components/eventing-controller/controllers/backend/reconciler.go:488]: do we really need to export such functions?

File: components/eventing-controller/api/v1alpha1/condition.go [hunk 2]
```diff
@@ -57,7 +57,9 @@ const (
 	ConditionReasonControllerStartFailed          ConditionReason = "Starting the controller failed"
 	ConditionReasonControllerStopFailed           ConditionReason = "Stopping the controller failed"
 	ConditionReasonPublisherProxySecretError      ConditionReason = "Publisher proxy secret sync failed"
-	ConditionDuplicateSecrets                     ConditionReason = "Multiple eventing backend labeled secrets exist"
+	//nolint:gosec // no potential hardcoded credentials
```
[the remark this answers is not in the document]
Comment (raypinto) [reply components/eventing-controller/api/v1alpha1/condition.go:60]: is it?

File: components/eventing-controller/controllers/backend/reconciler.go (hunk 1 above)
Comment (raypinto) [reply components/eventing-controller/controllers/backend/reconciler.go:488]: no. I will make them internal 👍

**Review by k15r — APPROVED** [review]

**Pull request merged**